src/references/RelationshipRef.test.ts

```diff
@@ -29,8 +29,19 @@ describe("RelationshipRef", () => {
 
         const queryResult = testClause.build();
         expect(queryResult.cypher).toMatchInlineSnapshot(`"this0this1"`);
+
         expect(rel1.type).toBe("ACTED_IN");
-        expect(rel1.type).toBeUndefined;
+        expect(rel2.type).toBeUndefined();
+    });
+
+    test("Create relationships with label expression", () => {
+        const rel1 = new Cypher.Relationship({ type: Cypher.labelExpr.or("ACTED_IN", "DIRECTED") });
+        const testClause = new TestClause(rel1);
+
+        const queryResult = testClause.build();
+        expect(queryResult.cypher).toMatchInlineSnapshot(`"this0"`);
+
+        expect(rel1.type).toEqual({ labels: ["ACTED_IN", "DIRECTED"], operator: "|" });
```

Comment:
If there's a better way to test this, let me know and I'll update it.

Reply:
:thinking: The only problem with this approach is that it needs to know the internals of the class `LabelExpr`.

Maybe we could test it with a `Pattern` which should generate a cypher containing the actual label `this0:ACTED_IN|DIRECTED`. It would be more of an integration test than a unit test, but that's where the labelExpr is really used.

Thoughts?

Reply:
I agree this is knowing a bit too much about the internals. I'm still very new to this library. How would you best use a `Pattern` here? It looks like it needs a NodeRef.

I was actually a little surprised that the output of `TestClause.build()` is just the variable and not the type string.

Maybe this isn't the right place for this test? Would label-expressions.test.ts be a better spot for it? You've got a test on unnamed Relationship there.

Reply:
Hi @jhanggi
That's fine :smile: 

`TestClause` is just a testing helper to be able to build any element in the cypher-builder. It returns the name of the variable because that's how variables are normally built.

Generating the labels is actually done by the [Pattern](https://neo4j.github.io/cypher-builder/cypher-builder/current/patterns/), so you'd need to create the pattern and then build it with the `TestClause`, for example:

```js
const pattern1 = new Cypher.Pattern(new Cypher.Node({})).related(rel1).to(new Cypher.Node({}))
new TestClause(pattern1).build()
```
We need to add a couple of extra nodes we don't really care about (that's because a pattern cannot be done as a standalone relationship without nodes)

You are probably right about this being the place for the test. In truth there are very few unit tests in the cypher builder, most of them are more akin to integration, so sometimes the organization is a bit tricky.

All of this said, the tests that you made (closer to unit tests) are also valid, if you find that testing the pattern (either in patterns.test.ts or label-expressions.test.ts or here itself) is too far removed from the actual problem you have fixed

Reply:
This does feel like the right place to test this, particularly that we can use these types, while the label expression test feels like the better place to really exercise all the `or`s and `and`s.

What would you think of introducing something like this. It hides away the node creation and focuses on the relationship piece.

```typescript
class TestRelClause {
    constructor(private rel: Cypher.Relationship) {}

    build() {
        const pattern = new Cypher.Pattern(new NodeRef()).related(this.rel).to(new NodeRef());
        return new TestClause(pattern).build();
    }
}

    test("Create relationships with label expression", () => {
        const rel1 = new Cypher.Relationship({ type: Cypher.labelExpr.or("ACTED_IN", "DIRECTED") });

        const testClause = new TestRelClause(rel1);

        const queryResult = testClause.build();
        expect(queryResult.cypher).toContain("[this0:(ACTED_IN|DIRECTED)]");
    });
```

Reply:
Mmm, it looks like a good idea but it is a bit overkill for a single test isn't it?

I'm not too against your original proposal, at the end of the day it does test the change you made. If you feel like using the pattern is too roundabout we can keep your original proposal

Reply:
I do think using the Pattern is a little roundabout. The original does have the disadvantage of knowing maybe a bit too much. Ultimately, what we're testing is that a type can be used. I would vote for keeping it simple.